Modules/_functoolsmodule.c

```diff
@@ -253,8 +253,13 @@ partial_repr(partialobject *pto)
     /* Pack keyword arguments */
     assert (PyDict_Check(pto->kw));
     for (i = 0; PyDict_Next(pto->kw, &i, &key, &value);) {
-        Py_SETREF(arglist, PyUnicode_FromFormat("%U, %U=%R", arglist,
+        /* Prevent key.__str__ from deleting key or value during formatting. */
+        Py_INCREF(key);
```

Comment:
Yes, indeed. I have no problems without that `INCREF` and I will remove that. That part was copied from [dict_repr](https://github.com/python/cpython/blob/master/Objects/dictobject.c#L2051) because I thought what works there needs to work here as well.